src/storage/core.rs

```diff
@@ -413,54 +416,69 @@ where
     /// Returns entries with matching key
     /// # Errors
     /// Fails after any disk IO errors.
-    pub async fn read_all(&self, key: impl AsRef<K>) -> Result<Vec<Entry>> {
+    pub async fn read_all(&self, key: impl AsRef<K>) -> Result<ReadResult<Vec<Entry>>> {
         let key = key.as_ref();
         let mut all_entries = Vec::new();
         let safe = self.inner.safe.read().await;
-        let active_blob = safe
-            .active_blob
-            .as_ref()
-            .ok_or_else(Error::active_blob_not_set)?;
-        if let Some(entries) = active_blob.read().await.read_all_entries(key).await? {
-            debug!(
-                "storage core read all active blob entries {}",
-                entries.len()
-            );
-            all_entries.extend(entries);
-        }
         let blobs = safe.blobs.read().await;
         let entries_closed_blobs = blobs
-            .iter_possible_childs(key)
+            .iter_possible_childs_rev(key)
             .map(|b| b.1.data.read_all_entries(key))
             .collect::<FuturesUnordered<_>>();
         entries_closed_blobs
-            .try_filter_map(future::ok)
+            .try_filter_map(|r| async move { Ok(r.into_option()) })
             .try_for_each(|v| {
                 debug!("storage core read all closed blob {} entries", v.len());
                 all_entries.extend(v);
                 future::ok(())
             })
             .await?;
-        debug!("storage core read all total {} entries", all_entries.len());
-        Ok(all_entries)
+        debug!(
+            "storage core read from non-active total {} entries",
+            all_entries.len()
+        );
+        let active_blob = safe
+            .active_blob
+            .as_ref()
+            .ok_or_else(Error::active_blob_not_set)?;
+        let read = active_blob.read().await.read_all_entries(key).await?;
+        if read.is_presented() {
+            let entries = read.unwrap();
+            debug!(
+                "storage core read all active blob entries {}",
+                entries.len()
+            );
+            all_entries.extend(entries);
```

Comment:
According to the latest changes in #209 the newest records are at the head of vector, so the `active_blob` should be processed first